analyses/snv-callers/util/wrangle_functions.R

```diff
@@ -0,0 +1,260 @@
+
+      # Create a categorical portion of the PolyPhen score
+      PolyPhen_category = stringr::word(PolyPhen, 1, sep = "\\("),
+
+      Variant_Classification = as.factor(Variant_Classification)
+    ) %>%
+    dplyr::mutate(
+      # From the base_change variable, summarize insertions, deletions, and
+      # changes that are more than one base into their own groups.
+      change = dplyr::case_when(
+        grepl("^-", base_change) ~ "ins",
+        grepl("-$", base_change) ~ "del",
+        nchar(base_change) > 3 ~ "long_change",
+        TRUE ~ base_change
+      )
+    ) %>%
+    dplyr::mutate(
+      # Create the mutation id based on the change variable as well as the
+      # gene symbol, start position, and sample ID.
+      mutation_id = paste0(
+        Hugo_Symbol, "_",
+        change, "_",
+        Start_Position, "_",
+        Tumor_Sample_Barcode
+      ),
+    )
+}
+
+maf_to_granges <- function(maf_df) {
+  # Turn MAF data.frame into a GRanges object. All of the original data.frame will
+  # be stored in the `mcols` slot of the GRanges object.
+  #
+  # Args:
+  #   maf_df: A MAF formatted data.frame with `Chromosome`, `Start_Position`,
+  #           `End_Position`, and `Strand`.
+  #
+  # Create the GRanges object
+  GenomicRanges::GRanges(
+    seqnames = maf_df$Chromosome,
+    ranges = IRanges::IRanges(
+      start = maf_df$Start_Position,
+      end = maf_df$End_Position
+    ),
+    strand = maf_df$Strand,
+    mcols = maf_df
+  )
+}
+
+wxs_bed_filter <- function(maf_df, wxs_bed = NULL, bp_window = 0) {
+  # Given a MAF formatted data.frame and a BED regions data.frame; filter out
+  # any variants of the MAF df that are not within the BED regions.
+  #
+  # Args:
+  #   maf_df: maf data that has been turned into a data.frame. Can be a maf object
+  #           that is subsetted using `@data`.
+  #   wxs_bed: a data.frame that has the windows to be used for TMB calculation.
+  #            BED formatted columns with chromosome, start, end positions in
+  #            that order.
+  #   bp_window: how many base pairs away can it be from the BED region to still
+  #              be included? Default is 0 bp. This argument gets forwarded
+  #              to GenomicRanges::findOverlaps's `maxgap` argument.
+
+  # Turn the WXS bed regions into a GRanges object
+  wxs_bed_granges <- GenomicRanges::GRanges(
+    seqnames = wxs_bed$X1,
+    ranges = IRanges::IRanges(
+      start = wxs_bed$X2,
+      end = wxs_bed$X3
+    )
+  )
+
+  # Obtain a MAF data.frame of only the WXS samples
+  maf_wxs <- maf_df %>%
+    dplyr::filter(experimental_strategy == "WXS")
+
+  # Error catcher
+  if (nrow(maf_wxs) == 0) {
+    warning("No WXS samples found underneath column 'experimental_strategy'
+            double check filtering steps and data.")
+  }
+
+  # Find the overlap of the BED regions and the mutations.
+  wxs_maf_granges <- GenomicRanges::GRanges(
```

Comment:
Can you explain why you're not using `maf_to_granges` here?

Reply:
No reason besides I forgot about my own function I made for this.